Test Utilities/Doubles/Reducer.swift

```diff
@@ -0,0 +1,48 @@
+import XCTest
+@testable import PusherChatkit
+
+public class DummyReducer<ActionType: Action, StateType: State>: DummyBase {
+    
+    public override init(file: StaticString = #file, line: UInt = #line) {
+        super.init(file: file, line: line)
+    }
+    
+    public func reducer(action: ActionType, state: StateType) -> StateType {
+        DummyFail(sender: self, function: #function)
+        return state
+    }
+    
+}
+
+public class StubReducer<ActionType: Action, StateType: State>: StubBase {
+    
+    private var reducer_expectedCallCount: UInt
+    private var reducer_expectedState: StateType
+    public private(set) var reducer_actionLastReceived: ActionType?
+    public private(set) var reducer_stateLastReceived: StateType?
+    public private(set) var reducer_actualCallCount: UInt = 0
+    
+    public init(reducer_expectedState: StateType,
+                reducer_expectedCallCount: UInt = 0,
+                file: StaticString = #file, line: UInt = #line) {
+        
+        self.reducer_expectedState = reducer_expectedState
```

Comment:
`reducer_expectedState` is never used and can be removed.

Reply:
Sorry, I am wrong, it is used.  I think it was the variable name that threw me as I thought you were checking this value against the _incoming_ state.  

I have been using `<func_name>_<object>ToReturn` in other Stubs, so for consistency perhaps you might rename to `reducer_stateToReturn`?

Reply:
Done